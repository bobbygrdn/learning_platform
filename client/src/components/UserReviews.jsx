import React from 'react'

export default function UserReviews() {
    // TODO: Finish user reviews component
    return (
        <div>

        </div>
    )
}
