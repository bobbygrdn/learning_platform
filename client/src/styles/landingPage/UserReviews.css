.reviews {
    display: flex;
    flex-direction: column;
    width: 80%;
    margin-bottom: 2em;
    pointer-events: none;
}