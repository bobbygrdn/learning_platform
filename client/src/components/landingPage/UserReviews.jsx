import React from 'react';
import Review from './Review';
import '../../styles/landingPage/UserReviews.css';
export default function UserReviews({ reviews }) {
    return (
        <div className='reviews' data-testid="reviews">
            {reviews ? reviews.map((review, index) => {
                return <Review key={index} review={review} />
            }) : null}
        </div>
    )
}
