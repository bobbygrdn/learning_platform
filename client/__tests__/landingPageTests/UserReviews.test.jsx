import React from 'react';
import { render, screen } from '@testing-library/react';
import UserReviews from '../../src/components/landingPage/UserReviews';

const reviews = () => {
    return [
        {
            id: 1,
            name: 'John Doe',
            review: 'Great place',
            rating: 5
        },
        {
            id: 2,
            name: 'Jane Doe',
            review: 'Amazing place',
            rating: 4
        },
        {
            id: 3,
            name: 'John Doe',
            review: 'Amazing place',
            rating: 5
        },
        {
            id: 4,
            name: 'John Doe',
            review: 'Amazing place',
            rating: 5
        },
        {
            id: 5,
            name: 'John Doe',
            review: 'Amazing place',
            rating: 5
        }
    ]
}

it('renders without crashing', () => {
    const { container } = render(<UserReviews reviews={reviews()} />);
    expect(container).toMatchSnapshot();
})

it('renders the same number of reviews as the number of reviews passed in', () => {
    render(<UserReviews reviews={reviews()} />);
    expect(screen.getAllByTestId('reviewText')).toHaveLength(reviews().length);
})