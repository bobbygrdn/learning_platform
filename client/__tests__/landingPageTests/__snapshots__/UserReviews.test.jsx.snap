// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="reviews"
    data-testid="reviews"
  >
    <q
      class="reviewText"
      data-testid="reviewText"
    />
    <p
      class="user"
      data-testid="user"
    />
    <q
      class="reviewText"
      data-testid="reviewText"
    />
    <p
      class="user"
      data-testid="user"
    />
    <q
      class="reviewText"
      data-testid="reviewText"
    />
    <p
      class="user"
      data-testid="user"
    />
    <q
      class="reviewText"
      data-testid="reviewText"
    />
    <p
      class="user"
      data-testid="user"
    />
    <q
      class="reviewText"
      data-testid="reviewText"
    />
    <p
      class="user"
      data-testid="user"
    />
  </div>
</div>
`;
